lantiq: move dsl related base-files into own package

sch-m opened the pull request:
It does not make sense to install this components on lantiq systems
where the dsl subsystem is not needed/used.

This also makes it possible to use the files also on other targets.
(hopefully ipq401x / FritzBox 7530 in the near future)

This commit is a part of "VRX518 support on ipq40xx" work you can find here:
https://github.com/3headeddevs/openwrt/tree/vrx518_support

Signed-off-by: Martin Schiller <ms.3headeddevs@gmail.com>

sch-m committed:
lantiq: move dsl related base-files into own package

It does not make sense to install this components on lantiq systems
where the dsl subsystem is not needed/used.

This also makes it possible to use the files also on other targets.
(hopefully ipq401x / FritzBox 7530 in the near future)

Signed-off-by: Martin Schiller <ms.3headeddevs@gmail.com>

adrianschmutzler commented on package/network/utils/ltq-dsl-base-files/Makefile:
```diff
@@ -0,0 +1,34 @@
+#
+# Copyright (C) 2019 OpenWrt.org
```
You are not affiliated with OpenWrt, so AFAIK you cannot put an OpenWrt copyright here.

sch-m replied:
OK, I'll remove that line.

adrianschmutzler replied:
Either put your name there or just omit the copyright and just keep the license.

sch-m replied:
Done

adrianschmutzler commented on package/network/utils/ltq-dsl-base-files/Makefile:
```diff
@@ -0,0 +1,34 @@
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=ltq-dsl-base-files
```
I'd prefer just "ltq-dsl-base"

sch-m replied:
I have chosen the name in reference to the origin of the files it contains, namely /target/linux/lantiq/base-files.

adrianschmutzler replied:
It's nitpicking, but essentially this is about what this package is supposed to evolve into:

ltq-dsl-base: This means it's the "base package" for the other two packages, and may contain whatever these packages would reasonably share (e.g. one could also move dependencies, setting kernel symbols etc.)

ltq-dsl-base-files: This means the package is limited to containing "files", i.e. shared scripts and library files.

Of course, this is nitpicking, and most probably nobody except me will really care about that :-)

sch-m replied:
OK, I agree.

adrianschmutzler replied:
Now, after I've finally offered you the option to choose :-)

sch-m replied:
I agree with you because I like the "not just files" argument.

adrianschmutzler commented:
I like this; one could think about renaming the files more appropriately, but I would _not_ do this in this first step, but, if at all, separately afterwards.

adrianschmutzler commented:
I plan to merge this, I will just leave it open for a few days to give other people the chance to comment/NAK.

adrianschmutzler commented:
BTW, I'm not really sure whether this should be in "utils" or in "config" subfolder like the other packages, but I don't really have much experience there. Maybe somebody can comment on that.

adrianschmutzler commented:
> . /usr/share/libubox/jshn.sh

Though it won't matter much practically, the package should add a dependency for that file. Think it would just be `+jshn`

adrianschmutzler commented:
> [ -f /lib/functions/lantiq_dsl.sh ] && . /lib/functions/lantiq_dsl.sh

The -f check can be removed for both packages as we have this file in ltq-dsl-base and that one is selected.

Would you provide that in a second patch and just push it to this PR? (Otherwise I'd do so when merging this ...)

https://github.com/openwrt/openwrt/blob/master/package/network/config/ltq-adsl-app/files/dsl_control#L11
https://github.com/openwrt/openwrt/blob/master/package/network/config/ltq-vdsl-app/files/dsl_control#L11

sch-m committed:
ltq-*dsl-app: dsl_control: remove unneeded check for lantiq_dsl.sh

This file is always present because it is part of the ltq-dsl-base
package on which these packages depend.

This check would not have been necessary in the past, because the script
was part of the TARGET_LANTIG on which these packages also depend.

Signed-off-by: Martin Schiller <ms@dev.tdt.de>

sch-m commented:
> > . /usr/share/libubox/jshn.sh
> 
> Though it won't matter much practically, the package should add a dependency for that file. Think it would just be `+jshn`

OK, I'll add this dependency.

sch-m commented:
> > [ -f /lib/functions/lantiq_dsl.sh ] && . /lib/functions/lantiq_dsl.sh
> 
> The -f check can be removed for both packages as we have this file in ltq-dsl-base and that one is selected.

OK, I'll do so, but this check would not have been necessary in the past either, because the script "lantiq_dsl.sh"
was part of TARGET_LANTIG on which these packages also depend.

adrianschmutzler commented:
Thanks, I've just merged the PR.

@feckert You should update your patch https://patchwork.ozlabs.org/project/openwrt/patch/20200616082613.892-1-fe@dev.tdt.de/ now. Note that I won't merge that one lacking the background to properly review it, I just commented on the ids back then based on "common sense". But thanks for pushing this one here forward.
